examples/resources/powerflex_service/resource.tf

```
@@ -32,3 +34,20 @@ resource "powerflex_service" "service" {
   clone_from_host        = "pfmc-k8s-20230809-160"
   deployment_timeout     = 0
 }
+
+
+# example 2: Get Template ID and Firmware ID from Template Datasource
+
+data "powerflex_template" "template_data" {
+  template_names = ["Teample_Name"]
+}
+
+resource "powerflex_service" "service" {
+  deployment_name        = "Test-Create-U1"
+  deployment_description = "Test Service-U1"
+  template_id            = data.powerflex_template.template_data.template_details[0].id
+  firmware_id            = data.powerflex_template.template_data.template_details[0].firmware_id
```

```suggestion
  template_id            = tolist(data.powerflex_template.template_data.template_details)[0].id
  firmware_id            = tolist(data.powerflex_template.template_data.template_details)[0].firmware_id
```
Set datastructure cannot be indexed directly